src/cloudHandler.js

```diff
@@ -11,7 +14,11 @@ var handler = function (params, success, failure) {
         cloudPath = '/sync/';
     }
     var url = cloudURL + cloudPath + params.dataset_id;
-    var json = JSON.stringify(params.req);
+    var payload = params.req;
+    payload.__fh = {
```

Comment:
Really good point! I wasn't sure about this as I want to maintain fully compatibility with fh-js-sdk and mbaas-api. In fh-js-sdk this client id is applied indirectly as part of the fh.cloud call.  This part is just drop in upstream replacement for cloud call and developers will look into this class to provide some custom implementations - for example React Native.
If we choose to do some refactoring on js-sdk then my work will probably need to be part of the product plan. For the moment I'm doing stuff after hours with agreement that changes will not affect our product etc.

Maybe we can have quick chat about this to make sure that we taking the best aproach.